### PR TITLE
bundle fee withdrawal (#419)

### PR DESCRIPTION
**Pull request by doerfli**
- [x] testcase: withdraw bundle fee
- [x] testcase: withdraw bundle fee from locked bundle (must work)
- [x] testcase: withdraw bundle fee not investor
- [x] testcase: withdraw bundle fee with max amount
- [x] testcase: withdraw bundle fee amount > available fee
- [x] testcase: withdraw bundle fee allowance too small
- [x] testcase: withdraw bundle fee amount = 0

### CODE REVIEW

File: contracts/distribution/DistributionService.sol [hunk 1]
```diff
@@ -276,14 +273,10 @@ contract DistributionService is
...
         IDistribution.DistributorInfo memory distributorInfo = reader.getDistributorInfo(distributorNftId);
-
-        if (!distributorInfo.active) {
-            revert ErrorDistributionServiceDistributorNotActive(distributorNftId);
-        }
         
         // determine withdrawn amount
         withdrawnAmount = amount;
-        if (withdrawnAmount.eq(AmountLib.max())) {
+        if (withdrawnAmount.gte(AmountLib.max())) {
```
Comment (matthiaszimmermann) [inline contracts/distribution/DistributionService.sol:279]: why gte and not eq? how could that happen?

File: contracts/pool/BundleService.sol [hunk 2]
```diff
@@ -28,6 +31,7 @@ contract BundleService is
     IBundleService 
 {
     using NftIdLib for NftId;
```
Comment (matthiaszimmermann) [inline contracts/pool/BundleService.sol:33]: all this using x for y is not necessary

File: contracts/pool/BundleService.sol [hunk 3]
```diff
@@ -262,6 +266,60 @@ contract BundleService is
...
+        InstanceReader reader = instance.getInstanceReader();
+        
+        IComponents.ComponentInfo memory poolInfo = reader.getComponentInfo(poolNftId);
+        address poolWallet = poolInfo.wallet;
+        
+        IBundle.BundleInfo memory bundleInfo = reader.getBundleInfo(bundleNftId);
+        
+        // determine withdrawn amount
+        withdrawnAmount = amount;
+        if (withdrawnAmount.gte(AmountLib.max())) {
```
Comment (matthiaszimmermann) [inline contracts/pool/BundleService.sol:286]: same as above, why gte?

File: contracts/pool/BundleService.sol [hunk 4]
```diff
@@ -262,6 +266,60 @@ contract BundleService is
...
+            if (withdrawnAmount.gt(reader.getFeeAmount(bundleNftId))) {
+                revert ErrorBundleServiceFeesWithdrawAmountExceedsLimit(withdrawnAmount, reader.getFeeAmount(bundleNftId));
+            }
+        }
+
+        if (withdrawnAmount.eqz()) {
+            revert ErrorBundleServiceFeesWithdrawAmountIsZero();
+        }
+
+        // check allowance
```
Comment (matthiaszimmermann) [inline contracts/pool/BundleService.sol:298]: do we have a ticket that says that we want/need to centralize this functionality? maybe into a lib?

File: contracts/pool/BundleService.sol (hunk 2 above)
Comment (doerfli) [reply contracts/pool/BundleService.sol]: fixed

File: contracts/pool/BundleService.sol (hunk 4 above)
Comment (doerfli) [reply contracts/pool/BundleService.sol:296]: [#417](https://github.com/etherisc/gif-next/issues/417)

**Pull request merged**